ui/components/app/confirm-page-container/confirm-page-container-content/confirm-page-container-content.component.js

```diff
@@ -117,15 +124,32 @@ export default class ConfirmPageContainerContent extends Component {
         {this.renderContent()}
         {(errorKey || errorMessage) && (
           <div className="confirm-page-container-content__error-container">
-            <ErrorMessage errorMessage={errorMessage} errorKey={errorKey} />
+            <ErrorMessage
+              errorMessage={this.context.t('somethingWentWrong')}
+              errorKey={errorKey}
```

Comment:
In the case where `errorKey` is set, this component is broken.

`errorKey`, when set, is used to obtain the error message. This is a deprecated pattern that we're trying to get rid of because it makes these keys impossible to find with strict-mode of the localized message linter, but nonetheless it's something we need to continue to support until we remove all uses.

Right now if it's set, the full error message will be shown (instead of "something went wrong") along with the "more details" button. When the user clicks "more details", they'll be shown an empty modal.

We need to decide - do we want to treat keyed errors the same way we do text errors? If so, we need to stop passing it into `ErrorMessage` here, and use the key to get the full error message below when passing it into `TransactionErrorDetailsModal`.

If we want to treat keyed errors differently than raw string errors (because they have translations), we can show them directly here and omit the "More details" link altogether.

Reply:
@Gudahtt What do you about about splitting this component, so we have one with the ability to accept links and the other one without?

Reply:
That could work! They seem different enough - one with plain text (the error key case) and one with a generic error with a more details button. Good idea.

Reply:
cool

Reply:
The second new component looks great!